homeassistant/components/garmin_connect/sensor.py

```diff
@@ -0,0 +1,255 @@
+    "latestSpo2ReadingTimeLocal": ["Latest SPO2 Time", "", "mdi:diabetes"],
+    "averageMonitoringEnvironmentAltitude": [
+        "Average Altitude",
+        "%",
+        "mdi:image-filter-hdr",
+    ],
+    "durationInMilliseconds": ["Duration", "ms", "mdi:progress-clock"],
+    "highestRespirationValue": ["Highest Respiration", "brpm", "mdi:progress-clock"],
+    "lowestRespirationValue": ["Lowest Respiration", "brpm", "mdi:progress-clock"],
+    "latestRespirationValue": ["Latest Respiration", "brpm", "mdi:progress-clock"],
+    "latestRespirationTimeGMT": ["Latest Respiration Update", "", "mdi:progress-clock"],
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+GARMIN_DEFAULT_CONDITIONS = ["totalSteps"]
+ATTRIBUTION = "Data provided by garmin.com"
+DEFAULT_NAME = "Garmin"
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_EMAIL): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(
+            CONF_MONITORED_CONDITIONS, default=GARMIN_DEFAULT_CONDITIONS
+        ): vol.All(cv.ensure_list, [vol.In(GARMIN_CONDITIONS_LIST)]),
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Garmin Connect component."""
+
+    email = config.get(CONF_EMAIL)
+    password = config.get(CONF_PASSWORD)
+    prefix_name = config.get(CONF_NAME)
+
+    try:
+        garmin_client = garminconnect.Garmin(email, password)
+    except ValueError as err:
+        _LOGGER.error("Error occured during Garmin Connect Client init: %s", err)
+        return
+
+    garmin_data = GarminConnectClient(garmin_client)
+
+    entities = []
+    for resource in config[CONF_MONITORED_CONDITIONS]:
+        sensor_type = resource
+        name = prefix_name + " " + GARMIN_CONDITIONS_LIST[resource][0]
+        unit = GARMIN_CONDITIONS_LIST[resource][1]
+        icon = GARMIN_CONDITIONS_LIST[resource][2]
+
+        _LOGGER.debug(
+            "Registered new sensor: %s, %s, %s, %s", sensor_type, name, unit, icon
+        )
+        entities.append(GarminConnectSensor(garmin_data, sensor_type, name, unit, icon))
+
+    add_entities(entities, True)
+
+
+class GarminConnectClient:
+    """Set up the Garmin Connect client."""
+
+    def __init__(self, client):
+        """Initialize the client."""
+        self.client = client
+        self.data = None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Fetch the latest data."""
+        today = date.today()
+        try:
+            self.data = self.client.fetch_stats(today.strftime("%Y-%m-%d"))
+        except ValueError as err:
+            _LOGGER.error("Error occured while fetching Garmin Connect data: %s", err)
+            return
+
+
+class GarminConnectSensor(Entity):
+    """Representation of a Garmin Connect Sensor."""
+
+    def __init__(self, data, sensor_type, name, unit, icon):
```

Comment:
Do you have data to provide a unique ID? Maybe something as a device ID + sensor_type would be unique ?

Reply:
I use users full name as filled in in Garmin Connect profile.